internal/app/oauth/storage.go

```diff
@@ -0,0 +1,337 @@
+package oauth
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"math/big"
+	mRand "math/rand"
+	"net"
+	"time"
+
+	"github.com/caos/oidc/pkg/oidc"
+	"github.com/caos/oidc/pkg/op"
+	"github.com/oklog/ulid"
+	"gopkg.in/square/go-jose.v2"
+)
+
+var (
+	appConfig *AuthSrvConfig
+	clients   map[string]op.Client
+	requests  map[string]AuthRequest
+	keySet    jose.JSONWebKeySet
+)
+
+type AuthStorage struct {
```

Comment:
```suggestion
type Storage struct {
```

package name is already `oauth` (my suggestion is `appauth`) so `Auth` prefix is not needed.

You can see this type as following:

```go
oauth.AuthStorage{...} // not good
oauth.Storage{...} // good: ok, it's a storage of oauth
```